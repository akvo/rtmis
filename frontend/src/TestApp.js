import { MemoryRouter } from "react-router-dom";
import { CookiesProvider } from "react-cookie";
import App from "./App";

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

const TestApp = ({ entryPoint = "/" }) => {
  return (
    <CookiesProvider>
      <MemoryRouter initialEntries={[entryPoint]}>
        <App />
      </MemoryRouter>
    </CookiesProvider>
  );
};

export default TestApp;
