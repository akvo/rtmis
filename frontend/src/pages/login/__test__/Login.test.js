import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import userEvent from "@testing-library/user-event";
import axios from "axios";
import TestApp from "../../../TestApp";
import "@testing-library/jest-dom";

jest.mock("axios");

describe("Login and Registration", () => {
  test("test if the login form exists", () => {
    const { asFragment } = render(<TestApp />);
    userEvent.click(screen.getByText(/Log In/i), { button: 0 });
    expect(screen.getByText(/Welcome back/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Please enter your account details/i)
    ).toBeInTheDocument();
    expect(screen.getByText(/Email Address/i)).toBeInTheDocument();
    expect(screen.getByText(/Forgot password/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot("LoginPage");
  });

  test("test if the registration form exists", async () => {
    const fakeUser = {
      name: "John Doe",
      invite: "abcd",
    };
    axios.mockResolvedValue({ status: 200, data: fakeUser });

    let registrationPage;
    await act(async () => {
      registrationPage = render(<TestApp entryPoint={"/login/abcd"} />);
      expect(screen.getByText(/Invalid/i)).toBeInTheDocument();
    });

    const welcome = screen.getByTestId("welcome-title");
    expect(welcome.textContent).toBe(
      `Welcome to RUSH, ${fakeUser.name}Please set your password for the platform.Your password must include:`
    );

    expect(screen.getByText(/Confirm Password/i)).toBeInTheDocument();
    expect(screen.getByText(/Set New Password/i)).toBeInTheDocument();
    expect(registrationPage.asFragment()).toMatchSnapshot("RegistrationPage");
  });
});
