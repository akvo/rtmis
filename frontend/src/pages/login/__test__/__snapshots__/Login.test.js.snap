// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login and Registration test if the login form exists: LoginPage 1`] = `
<DocumentFragment>
  <div
    class="ant-row  layout"
  >
    <div
      class="body"
    >
      <div
        id="login"
      >
        <div
          class="background"
          style="background-image: url(/assets/banner.png); background-size: cover; background-position: center; background-repeat: no-repeat;"
        />
        <div
          class="logo"
        >
          <a
            href="/"
          >
            <img
              alt="/logo.png"
              src="/logo.png"
            />
            <h1>
              Ministry of Health
            </h1>
          </a>
        </div>
        <div
          class="ant-row ant-row-middle wrapper"
        >
          <div
            class="ant-col ant-col-12 left-side"
          >
            <div
              class="title"
            >
              <h1>
                Welcome to the National
                <br />
                Sanitation and Hygiene
                <br />
                Real-Time Monitoring System
              </h1>
            </div>
          </div>
          <div
            class="ant-col ant-col-12 right-side"
          >
            <h1>
              Welcome back
              <br />
              <small>
                Please enter your account details
              </small>
            </h1>
            <form
              class="ant-form ant-form-vertical"
              id="login-form"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="login-form_email"
                    title="Email Address"
                  >
                    Email Address
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <span
                            aria-label="user"
                            class="anticon anticon-user site-form-item-icon"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="user"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          class="ant-input"
                          id="login-form_email"
                          placeholder="Email"
                          type="text"
                          value=""
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="login-form_password"
                    title="Password"
                  >
                    Password
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <span
                        class="ant-input-affix-wrapper ant-input-password"
                      >
                        <span
                          class="ant-input-prefix"
                        >
                          <span
                            aria-label="lock"
                            class="anticon anticon-lock site-form-item-icon"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="lock"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                              />
                            </svg>
                          </span>
                        </span>
                        <input
                          action="click"
                          class="ant-input"
                          id="login-form_password"
                          placeholder="Password"
                          type="password"
                          value=""
                        />
                        <span
                          class="ant-input-suffix"
                        >
                          <span
                            aria-label="eye-invisible"
                            class="anticon anticon-eye-invisible ant-input-password-icon"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="eye-invisible"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                              />
                              <path
                                d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                              />
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <label
                        class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
                      >
                        <span
                          class="ant-checkbox ant-checkbox-checked"
                        >
                          <input
                            checked=""
                            class="ant-checkbox-input"
                            id="login-form_remember"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="ant-checkbox-inner"
                          />
                        </span>
                        <span>
                          Remember me
                        </span>
                      </label>
                      <a
                        class="login-form-forgot"
                        href="/forgot-password"
                      >
                        Forgot password
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          Log in
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Login and Registration test if the registration form exists: RegistrationPage 1`] = `
<DocumentFragment>
  <div
    class="ant-row  layout"
  >
    <div
      class="body"
    >
      <div
        id="login"
      >
        <div
          class="background"
          style="background-image: url(/assets/banner.png); background-size: cover; background-position: center; background-repeat: no-repeat;"
        />
        <div
          class="logo"
        >
          <a
            href="/"
          >
            <img
              alt="/logo.png"
              src="/logo.png"
            />
            <h1>
              Ministry of Health
            </h1>
          </a>
        </div>
        <div
          class="ant-row ant-row-middle wrapper"
        >
          <div
            class="ant-col ant-col-12 left-side"
          >
            <div
              class="title"
            >
              <h1>
                Welcome to the National
                <br />
                Sanitation and Hygiene
                <br />
                Real-Time Monitoring System
              </h1>
            </div>
          </div>
          <div
            class="ant-col ant-col-12 right-side"
          >
            <div>
              <h1
                data-testid="welcome-title"
              >
                Welcome to RTMIS, John Doe
                <br />
                <small>
                  Set your own password including the following criteria
                </small>
              </h1>
              <div
                class="ant-checkbox-group"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-group-item"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                      value="Lowercase Character"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    Lowercase Character
                  </span>
                </label>
                <label
                  class="ant-checkbox-wrapper ant-checkbox-group-item"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                      value="Numbers"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    Numbers
                  </span>
                </label>
                <label
                  class="ant-checkbox-wrapper ant-checkbox-group-item"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                      value="Special Character"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    Special Character
                  </span>
                </label>
                <label
                  class="ant-checkbox-wrapper ant-checkbox-group-item"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                      value="Uppercase Character"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    Uppercase Character
                  </span>
                </label>
                <label
                  class="ant-checkbox-wrapper ant-checkbox-group-item"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                      value="No White Space"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    No White Space
                  </span>
                </label>
                <label
                  class="ant-checkbox-wrapper ant-checkbox-group-item"
                >
                  <span
                    class="ant-checkbox"
                  >
                    <input
                      class="ant-checkbox-input"
                      type="checkbox"
                      value="Minimum 8 Character"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    Minimum 8 Character
                  </span>
                </label>
              </div>
              <form
                class="ant-form ant-form-vertical"
                id="registration-form"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="registration-form_password"
                      title="Password"
                    >
                      Password
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-input-affix-wrapper ant-input-password"
                        >
                          <span
                            class="ant-input-prefix"
                          >
                            <span
                              aria-label="user"
                              class="anticon anticon-user site-form-item-icon"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="user"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                                />
                              </svg>
                            </span>
                          </span>
                          <input
                            action="click"
                            class="ant-input"
                            id="registration-form_password"
                            placeholder="Password"
                            type="password"
                          />
                          <span
                            class="ant-input-suffix"
                          >
                            <span
                              aria-label="eye-invisible"
                              class="anticon anticon-eye-invisible ant-input-password-icon"
                              role="img"
                              tabindex="-1"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="eye-invisible"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                />
                                <path
                                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="registration-form_confirm"
                      title="Confirm Password"
                    >
                      Confirm Password
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-input-affix-wrapper ant-input-password"
                        >
                          <span
                            class="ant-input-prefix"
                          >
                            <span
                              aria-label="lock"
                              class="anticon anticon-lock site-form-item-icon"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="lock"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                                />
                              </svg>
                            </span>
                          </span>
                          <input
                            action="click"
                            class="ant-input"
                            id="registration-form_confirm"
                            placeholder="Confirm Password"
                            type="password"
                          />
                          <span
                            class="ant-input-suffix"
                          >
                            <span
                              aria-label="eye-invisible"
                              class="anticon anticon-eye-invisible ant-input-password-icon"
                              role="img"
                              tabindex="-1"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="eye-invisible"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                />
                                <path
                                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          type="submit"
                        >
                          <span>
                            Set New Password
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
